### PR TITLE
Port to pcre2

### LINKED ISSUE #52: Please port from PCRE to PCRE2
pcre has been replaced by its successor pcre2.
Please consider porting libbytesize to it.

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Fixes #52

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

File: src/Makefile.am [hunk 1]
```diff
@@ -3,7 +3,7 @@ LDADD = $(LIBINTL)
 
 lib_LTLIBRARIES = libbytesize.la
 libbytesize_la_CFLAGS = -Wall -Wextra -Werror -Wno-overflow -D_GNU_SOURCE
-libbytesize_la_LIBADD = -lgmp -lmpfr -lpcre
+libbytesize_la_LIBADD = -lgmp -lmpfr -lpcre2-8
```
Comment (vpodzime) [inline src/Makefile.am]: Can we use `$PCRE2-LIBS` or something like that here?

File: src/bs_size.c [hunk 2]
```diff
@@ -7,14 +7,18 @@
 #include <string.h>
 #include <ctype.h>
 #include <limits.h>
-#include <pcre.h>
+
+#define PCRE2_CODE_UNIT_WIDTH 8
```
Comment (vpodzime) [inline src/bs_size.c]: This definitely deserves a comment.

File: src/Makefile.am (hunk 1 above)
Comment (vojtechtrefny) [reply src/Makefile.am]: Fixed.

File: src/bs_size.c (hunk 2 above)
Comment (vojtechtrefny) [reply src/bs_size.c]: Fixed.

**Pull request merged**